src/provisioner/provisioner.py

```diff
@@ -168,15 +187,66 @@ def apply_cluster_config(cortx_conf, cortx_release):
 
             solution_config_nodes = CortxCluster(nodes, cortx_release)
             solution_config_nodes.save(cortx_conf)
-            solution_config_storagesets = CortxStorageSet(storage_sets)
-            solution_config_storagesets.save(cortx_conf)
+            Machine_id = CortxProvisioner._get_machine_id()
+            if not CortxProvisioner._wait_for_lock_to_be_released(cortx_conf, CortxProvisioner._timeout):
+                if not Conf.unlock(cortx_conf._conf_idx, lock_owner=Machine_id, force = True, lock_key=CortxProvisioner._lock_key):
+                    raise CortxProvisionerError(errno.EINVAL, f"Force unlock failed for index {cortx_conf._conf_idx}")
+                # TODO: remove Conf.save once gconf is completly moved to consul
+                Conf.save(cortx_conf._conf_idx)
+            if cortx_conf.get('cluster>id') is None:
+                cluster_keys = [('cluster>id', cluster_id),
+                    ('cluster>name', cluster_name)]
+                cortx_conf.set_kvs(cluster_keys)
+                if not Conf.lock(cortx_conf._conf_idx, lock_owner=Machine_id, lock_key=CortxProvisioner._lock_key):
+                    raise CortxProvisionerError(errno.EINVAL, f"locking failed for index {cortx_conf._conf_idx}")
+                # TODO: remove Conf.save once gconf is completly moved to consul
+                Conf.save(cortx_conf._conf_idx)
+                solution_config_storagesets = CortxStorageSet(storage_sets)
+                solution_config_storagesets.save(cortx_conf)
+                Conf.save(cortx_conf._conf_idx)
+                if not Conf.unlock(cortx_conf._conf_idx, lock_owner=Machine_id, lock_key=CortxProvisioner._lock_key):
+                    raise CortxProvisionerError(errno.EINVAL, f"unlocking failed for index {cortx_conf._conf_idx}")
+                # TODO: remove Conf.save once gconf is completly moved to consul
+                Conf.save(cortx_conf._conf_idx)
         except KeyError as e:
             raise CortxProvisionerError(
                 errno.EINVAL,
                 f'Error occurred while applying cluster_config {e}')
 
     @staticmethod
-    def _get_node_info(_conf_idx: str):
+    def _check_consul_connection(cortx_conf_url: str, timeout: int):
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock.settimeout(2)
```

Comment:
Do not hard-code time interval (either define constant value at top or accept it as input argument)

Reply:
done, removed hardcoded value